plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,655 @@
+
+
+### Operator tries to steal funds from an included transaction
+
+1. Alice spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is included in (valid) block `N`.
+3. Operator creates invalid deposit, creating `UTXO3`.
+4. Operator spends `UTXO3` in `TX3`, creating `UTXO4`.
+5. Operator starts an exit referencing `TX3` and places `exit bond`.
+6. Operator piggybacks onto the exit and places `piggyback bond`.
+7. Bob starts a *standard* exit for `UTXO2`.
+8. Operator’s exit will have priority of position of `UTXO3`.
+Bob’s exit will have priority of position of `UTXO2`.
+9. Bob receives the value of `UTXO2` first, Operator receives the value of `UTXO4` second.
+All bonds are refunded.
+
+
+### Operator tries to steal funds from an in-flight transaction.
+
+1. Alice spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is in-flight.
+3. Operator creates invalid deposit, creating `UTXO3`.
+4. Operator spends `UTXO3` in `TX3`, creating `UTXO4`.
+5. Operator starts an exit referencing `TX3` and places `exit bond`.
+6. Operator piggybacks onto the exit and places `piggyback bond`.
+7. Bob starts an exit referencing `TX1` and places `exit bond`.
+8. Bob piggybacks onto the exit and places `piggyback bond`.
+9. Alice is honest, so she hasn’t spent `UTXO1` in any transaction other than `TX1`.
+10. Operator’s exit will have priority of position of `UTXO3`.
+Bob’s exit will have priority of position of `UTXO1`.
+11. Bob receives the value of `UTXO2` first, Operator receives the value of `UTXO4` second.
+All bonds are refunded.
+
+
+### Attack Vectors and Mitigations
+
+#### Honest Exit Bond Slashing
+
+It’s possible for an honest user to start an exit and have their exit bond slashed.
+This can occur if one of the inputs to a transaction is malicious and signs a second transaction spending the same input.
+
+The following scenario demonstrates this attack:
+1. Mallory spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is in-flight.
+3. Operator begins withholding blocks while `TX1` is still in-flight.
+4. Bob starts an exit referencing `TX1` and places `exit bond`.
+5. Mallory spends `UTXO1` in `TX2`.
+6. In period 1 of the exit for `TX1`, Mallory challenges the canonicity of `TX1` by revealing `TX2`.
+7. No one is able to respond to the challenge in period 2, so `TX1` is determined to be non-canonical.
+8. After period 2, Mallory receives Bob’s `exit bond`, no one exits any UTXOs.
+
+Mallory has therefore caused Bob to lose `exit bond`, even though Bob was acting honestly.
+We want to mitigate the impact of this attack as much as possible so that this does not prevent users from receiving funds.
+ 
+
+##### Mitigations for Honest Exit Bond Slashing
+
+###### Bond Sharing
+
+One way to partially mitigate this attack is for each user who piggybacks to cover some portion of `exit bond`.
+This cuts the per-person value of `exit bond` proportionally to the number of users who have piggybacked.
```

Comment:
This wouldn't help the Bob in the example though.

Reply:
Would help Bob if other honest users piggybacked alongside Bob.

Reply:
Right but it's presenting a solution that doesn't necessarily address the problem as stated above. Could cause some confusion. Easy thing would be to add something like, "While it wouldn't mitigate the single output example above, a method for multiple input/output transaction would be..."